shader/fine.wgsl

```diff
@@ -265,6 +286,28 @@ fn main(
                 }
                 cmd_ix += 3u;
             }
+            // CMD_IMAGE
+            case 8u: {
+                let image = read_image(cmd_ix);
+                let atlas_extents = image.atlas_offset + image.extents;
+                for (var i = 0u; i < PIXELS_PER_THREAD; i += 1u) {
+                    let my_xy = vec2(xy.x + f32(i), xy.y);
+                    let atlas_uv = image.matrx.xy * my_xy.x + image.matrx.zw * my_xy.y + image.xlat + image.atlas_offset;
+                    // This currently clips to the image bounds. TODO: extend modes
+                    if all(atlas_uv < atlas_extents) {
```

Comment:
My gut feeling is that performance might be improved considerably if this is also predicated on `area[i] != 0.0`, though that will almost certainly depend on the hardware. These loads are already conditional, and I think it can reduce texture sample bandwidth for the tile wastage case. Perhaps it's worth doing careful experiments and landing this with the simplest case.

Reply:
We had discussed this and I agree that since we’re already branching, might as well add the additional check. Will do before merging.